### DISCUSSION TITLE
need to change highlighting for Comments, CursorLine, and a few others

### DISCUSSION THREAD

**Discussion by am-zest** [LazyVim/LazyVim#2022] [Q&A] [answered]
Hi,

I've just started with LazyVim today and it seems very nice so far.

I'm trying to customize the highlighting slightly. I'd like to achieve the effects of the follow commands:
```
vim.api.nvim_command("highlight Comment guifg=#888888")
vim.api.nvim_command("highlight WinSeparator guifg=#888888")
vim.api.nvim_command("highlight NvimTreeWinSeparator guifg=#888888")
vim.api.nvim_command("highlight CursorLine guibg=#333666 gui=underline")
vim.api.nvim_command("highlight NvimTreeCursorLine guibg=#333666")
vim.api.nvim_command("highlight TelescopeBorder guifg=#888888")
vim.api.nvim_command("highlight IndentBlanklineChar guifg=#ffffff")
```

how do I do that in LazyVim?

Thank you!

**Comment by dimas-cyriaco** [comment]
I'ld like to know the best practices for this too. Currently I'm putting these commands on the `config/lazy.lua` file bellow the `require("lazy").setup()` call. It works, but feels kinda hacky...

**Reply by dpetka2001** [reply, marked as answer]
Best way (in my opinion) would be to customize the highlights in the spec of the colorscheme you use. See [here](https://github.com/dpetka2001/dotfiles/blob/1ae238b4f9f070eebabe434290ed7aa1c01ab0fa/dot_config/nvim/lua/plugins/colorscheme.lua#L7-L42) how I do it in my dotfiles for the `tokyonight` colorscheme (although maybe that would depend from colorscheme to colorscheme, so would be best to read the docs to see how it allows the user to make changes). That's assuming that the colorscheme has support for the highlight groups you want to customize. If it doesn't then I guess I would put those lines (maybe??) in the `config` function of the colorscheme I use, for the changes to take effect after the colorscheme has loaded.

**Reply by dimas-cyriaco** [reply]
Thank you. I'm using `tokyonight` and putting the configs on the `on_highlights` option worked perfectly. 🎉 

Here's how it ended up:

```lua
return   {
    "folke/tokyonight.nvim",
    event = { "VeryLazy" },
    opts = {
      style = "storm",
      on_highlights = function(hl, c)
        hl.Folded = {
          bg = c.bg_highlight,
        }
      end,
    },
  }
```

**Reply by armyers** [reply]
@dpetka2001 and @dimas-cyriaco thank you for your responses! I won't be able to try them today, but they're very helpful. And the link to your (@dpetka2001) dot files will give me something to learn from. thank you!

**Reply by armyers** [reply]
yes, this worked for me as well. thank you!

```
return {
  "folke/tokyonight.nvim",
  event = { "VeryLazy" },
  opts = {
    style = "night",
    dim_inactive = true,
    on_highlights = function(hl, c)
      hl.CursorLine = {
        bg = "#333666",
        fg = "#bbbbbb",
        underline = true,
      }
      hl.NvimTreeCursorLine = {
        bg = "#333666",
        fg = "#bbbbbb",
      }
      hl.WinSeparator = {
        fg = "#555555",
      }
      hl.NvimTreeWinSeparator = {
        fg = "#555555",
      }
    end,
  },
}
```

**Comment by armyers** [comment]
sorry, I just realized this might be confusing, but armyers == am-zest. :)
